File: measurecompute/DAQ/C#/ULDI04/ULDI04.cs
// ==============================================================================
//
//  File:                         ULDI04.CS
//
//  Library Call Demonstrated:    MccDaq.MccBoard.DIn()
//
//  Purpose:                      Reads a value from MccDaq.DigitalPortType.AuxPort.
//
//  Demonstration:                Select number of bits and read
//                                    MccDaq.DigitalPortType.AuxPort.
//
//  Other Library Calls:          MccDaq.MccService.ErrHandling()
//								  MccDaq.MccBoard.DConfigPort()
//                                MccDaq.MccBoard.DioConfig.GetDevType()
//                                MccDaq.MccBoard.DioConfig.GetDInMask()
//                                MccDaq.MccBoard.DioConfig.GetDOutMask()
//
//  Special Requirements:         Board 0 must have a digital port.
//
// ==============================================================================
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Diagnostics;

using MccDaq;

namespace ULDI04
{
    public class frmDigAuxIn : System.Windows.Forms.Form
    {
     
        // Required by the Windows Form Designer
        private System.ComponentModel.IContainer components;

        public ToolTip ToolTip1;
        public Button cmdEndProgram;
        public Timer tmrReadInputs;
        public Label lblShowPortVal;
        public Label lblPortVal;
        public Label lblInstruct;
        public Label lblDemoFunction;

        private MccDaq.MccBoard DaqBoard;
        const MccDaq.DigitalPortType PortNum = MccDaq.DigitalPortType.AuxPort; //  set type of port to auxilliary


        public frmDigAuxIn()
        {
            
            // This call is required by the Windows Form Designer.
            InitializeComponent();

            //  Initiate error handling
            //   activating error handling will trap errors like
            //   bad channel numbers and non-configured conditions.
            //   Parameters:
            //     MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
            //     MccDaq.ErrorHandling.StopAll   :if an error is encountered, the program will stop
            
            MccDaq.ErrorInfo ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll);
            

            // Create a new MccBoard object for Board 0
            DaqBoard = new MccDaq.MccBoard(0);

            // Determine if board has AuxPort 
            //Parameters:
            //  devNum      : 0-based digital device index
            //  portType    : return value for type of device
            int devNum =0;  //AuxPort is the 0th digital device, if present
            int portType = 0;
            DaqBoard.DioConfig.GetDevType(devNum, out portType);
            if (portType != (int)MccDaq.DigitalPortType.AuxPort)
            {
                // this board doesn't have an AuxPort!
                Application.Exit();
            }

            // Check if AuxPort needs configuring 
            // Parameters:
            //     devNum      : 0-based digital device index
            //      mask    : bitmask indicating corresponding bit is in respecitve direction
            int inmask=0, outmask=0;
            DaqBoard.DioConfig.GetDInMask(devNum, out inmask);
            DaqBoard.DioConfig.GetDOutMask(devNum, out outmask);
            // only non-configurable AuxPorts have overlapping input and output masks
            if (0 == (inmask & outmask))
            {
                DaqBoard.DConfigPort(PortNum, MccDaq.DigitalPortDirection.DigitalIn);
            }

        }

        // Form overrides dispose to clean up the component list.
        protected override void  Dispose(bool Disposing)
        {
            if (Disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(Disposing);
        }

        
             
        
        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        
        private void InitializeComponent()
        {

            this.components = new System.ComponentModel.Container();
            this.ToolTip1 = new System.Windows.Forms.ToolTip(this.components);
            this.cmdEndProgram = new System.Windows.Forms.Button();
            this.tmrReadInputs = new System.Windows.Forms.Timer(this.components);
            this.lblShowPortVal = new System.Windows.Forms.Label();
            this.lblPortVal = new System.Windows.Forms.Label();
            this.lblInstruct = new System.Windows.Forms.Label();
            this.lblDemoFunction = new System.Windows.Forms.Label();
            this.SuspendLayout();
            //
            // cmdEndProgram
            //
            this.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control;
            this.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default;
            this.cmdEndProgram.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText;
            this.cmdEndProgram.Location = new System.Drawing.Point(256, 200);
            this.cmdEndProgram.Name = "cmdEndProgram";
            this.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No;
            this.cmdEndProgram.Size = new System.Drawing.Size(57, 33);
            this.cmdEndProgram.TabIndex = 3;
            this.cmdEndProgram.Text = "Quit";
            this.cmdEndProgram.Click += new System.EventHandler(this.cmdEndProgram_Click);
            //
            // tmrReadInputs
            //
            this.tmrReadInputs.Enabled = true;
            this.tmrReadInputs.Interval = 200;
            this.tmrReadInputs.Tick += new System.EventHandler(this.tmrReadInputs_Tick);
            //
            // lblShowPortVal
            //
            this.lblShowPortVal.BackColor = System.Drawing.SystemColors.Window;
            this.lblShowPortVal.Cursor = System.Windows.Forms.Cursors.Default;
            this.lblShowPortVal.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.lblShowPortVal.ForeColor = System.Drawing.Color.Blue;
            this.lblShowPortVal.Location = new System.Drawing.Point(272, 152);
            this.lblShowPortVal.Name = "lblShowPortVal";
            this.lblShowPortVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
            this.lblShowPortVal.Size = new System.Drawing.Size(33, 17);
            this.lblShowPortVal.TabIndex = 2;
            //
            // lblPortVal
            //
            this.lblPortVal.BackColor = System.Drawing.SystemColors.Window;
            this.lblPortVal.Cursor = System.Windows.Forms.Cursors.Default;
            this.lblPortVal.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.lblPortVal.ForeColor = System.Drawing.SystemColors.WindowText;
            this.lblPortVal.Location = new System.Drawing.Point(48, 152);
            this.lblPortVal.Name = "lblPortVal";
            this.lblPortVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
            this.lblPortVal.Size = new System.Drawing.Size(209, 17);
            this.lblPortVal.TabIndex = 1;
            //
            // lblInstruct
            //
            this.lblInstruct.BackColor = System.Drawing.SystemColors.Window;
            this.lblInstruct.Cursor = System.Windows.Forms.Cursors.Default;
            this.lblInstruct.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.lblInstruct.ForeColor = System.Drawing.Color.Red;
            this.lblInstruct.Location = new System.Drawing.Point(48, 72);
            this.lblInstruct.Name = "lblInstruct";
            this.lblInstruct.RightToLeft = System.Windows.Forms.RightToLeft.No;
            this.lblInstruct.Size = new System.Drawing.Size(233, 33);
            this.lblInstruct.TabIndex = 4;
            this.lblInstruct.Text = "Input a TTL high or low level to auxillary digital inputs to change Data Value.";
            this.lblInstruct.TextAlign = System.Drawing.ContentAlignment.TopCenter;
            //
            // lblDemoFunction
            //
            this.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window;
            this.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default;
            this.lblDemoFunction.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText;
            this.lblDemoFunction.Location = new System.Drawing.Point(40, 16);
            this.lblDemoFunction.Name = "lblDemoFunction";
            this.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No;
            this.lblDemoFunction.Size = new System.Drawing.Size(241, 41);
            this.lblDemoFunction.TabIndex = 0;
            this.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.DIn() and AuxPort";
            this.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter;
            //
            // frmDigAuxIn
            //
            this.AcceptButton = this.cmdEndProgram;
            this.AutoScaleBaseSize = new System.Drawing.Size(6, 13);
            this.BackColor = System.Drawing.SystemColors.Window;
            this.ClientSize = new System.Drawing.Size(331, 246);
            this.Controls.AddRange(new System.Windows.Forms.Control[] {this.cmdEndProgram, this.lblShowPortVal, this.lblPortVal, this.lblInstruct, this.lblDemoFunction});
            this.Font = new System.Drawing.Font("Arial", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.ForeColor = System.Drawing.SystemColors.WindowText;
            this.Location = new System.Drawing.Point(7, 103);
            this.Name = "frmDigAuxIn";
            this.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
            this.Text = "Universal Library Digital In (AUXPORT)";
            this.ResumeLayout(false);

        }

    #endregion

        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main() 
        {
            Application.Run(new frmDigAuxIn());
        }
        
        
        private void cmdEndProgram_Click(object eventSender, System.EventArgs eventArgs) /* Handles cmdEndProgram.Click */
        {
            Application.Exit();
        }


        private void tmrReadInputs_Tick(object eventSender, System.EventArgs eventArgs) /* Handles tmrReadInputs.Tick */
        {
            tmrReadInputs.Stop();

            ushort DataValue;
            MccDaq.ErrorInfo ULStat = DaqBoard.DIn( PortNum, out DataValue);

            lblPortVal.Text = "Value read at auxillary digital inputs:";
            lblShowPortVal.Text = DataValue.ToString("0");
          
            tmrReadInputs.Start();
        }

    }
}

File: measurecompute/DAQ/VB.NET/ULDI04.NET/ULDI04.vb
'==============================================================================

' File:                         ULDI04.VB

' Library Call Demonstrated:    MccDaq.MccBoard.DIn()

' Purpose:                      Reads a value from MccDaq.DigitalPortType.AuxPort.

' Demonstration:                Select number of bits and read 
'                                   MccDaq.DigitalPortType.AuxPort.

' Other Library Calls:          MccDaq.MccService.ErrHandling()
'                               MccDaq.MccBoard.DConfigPort()
'                               MccDaq.MccBoard.DioConfig.GetDevType()
'                               MccDaq.MccBoard.DioConfig.GetDInMask()
'                               MccDaq.MccBoard.DioConfig.GetDOutMask()

' Special Requirements:         Board 0 must have a digital port.

'==============================================================================
Option Strict Off
Option Explicit On 
Public Class frmDigAuxIn
    Inherits System.Windows.Forms.Form

    'Create a new MccBoard object for Board 0
    Private DaqBoard As MccDaq.MccBoard = New MccDaq.MccBoard(0)

    Const PortNum As MccDaq.DigitalPortType = MccDaq.DigitalPortType.AuxPort ' set type of port to auxilliary

    Private Sub cmdEndProgram_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdEndProgram.Click

        End

    End Sub

    Private Sub tmrReadInputs_Tick(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles tmrReadInputs.Tick
        Dim ULStat As MccDaq.ErrorInfo
        Dim DataValue As UInt16

        tmrReadInputs.Stop()
        
        ULStat = DaqBoard.DIn(PortNum, DataValue)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

        lblPortVal.Text = "Value read at auxillary digital inputs:"
        lblShowPortVal.Text = DataValue.ToString("0")
        
        tmrReadInputs.Start()

    End Sub

#Region "Windows Form Designer generated code "
    Public Sub New()
        MyBase.New()
        'This call is required by the Windows Form Designer.
        InitializeComponent()

        InitUL()

    End Sub
    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
        If Disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(Disposing)
    End Sub
    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer
    Public ToolTip1 As System.Windows.Forms.ToolTip
    Public WithEvents cmdEndProgram As System.Windows.Forms.Button
    Public WithEvents tmrReadInputs As System.Windows.Forms.Timer
    Public WithEvents lblShowPortVal As System.Windows.Forms.Label
    Public WithEvents lblPortVal As System.Windows.Forms.Label
    Public WithEvents lblInstruct As System.Windows.Forms.Label
    Public WithEvents lblDemoFunction As System.Windows.Forms.Label
    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container()
        Me.ToolTip1 = New System.Windows.Forms.ToolTip(Me.components)
        Me.cmdEndProgram = New System.Windows.Forms.Button()
        Me.tmrReadInputs = New System.Windows.Forms.Timer(Me.components)
        Me.lblShowPortVal = New System.Windows.Forms.Label()
        Me.lblPortVal = New System.Windows.Forms.Label()
        Me.lblInstruct = New System.Windows.Forms.Label()
        Me.lblDemoFunction = New System.Windows.Forms.Label()
        Me.SuspendLayout()
        '
        'cmdEndProgram
        '
        Me.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control
        Me.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default
        Me.cmdEndProgram.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText
        Me.cmdEndProgram.Location = New System.Drawing.Point(256, 200)
        Me.cmdEndProgram.Name = "cmdEndProgram"
        Me.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.cmdEndProgram.Size = New System.Drawing.Size(57, 33)
        Me.cmdEndProgram.TabIndex = 3
        Me.cmdEndProgram.Text = "Quit"
        '
        'tmrReadInputs
        '
        Me.tmrReadInputs.Enabled = True
        Me.tmrReadInputs.Interval = 200
        '
        'lblShowPortVal
        '
        Me.lblShowPortVal.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowPortVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowPortVal.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowPortVal.ForeColor = System.Drawing.Color.Blue
        Me.lblShowPortVal.Location = New System.Drawing.Point(272, 152)
        Me.lblShowPortVal.Name = "lblShowPortVal"
        Me.lblShowPortVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowPortVal.Size = New System.Drawing.Size(33, 17)
        Me.lblShowPortVal.TabIndex = 2
        '
        'lblPortVal
        '
        Me.lblPortVal.BackColor = System.Drawing.SystemColors.Window
        Me.lblPortVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblPortVal.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblPortVal.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblPortVal.Location = New System.Drawing.Point(48, 152)
        Me.lblPortVal.Name = "lblPortVal"
        Me.lblPortVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblPortVal.Size = New System.Drawing.Size(209, 17)
        Me.lblPortVal.TabIndex = 1
        '
        'lblInstruct
        '
        Me.lblInstruct.BackColor = System.Drawing.SystemColors.Window
        Me.lblInstruct.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblInstruct.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblInstruct.ForeColor = System.Drawing.Color.Red
        Me.lblInstruct.Location = New System.Drawing.Point(48, 72)
        Me.lblInstruct.Name = "lblInstruct"
        Me.lblInstruct.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblInstruct.Size = New System.Drawing.Size(233, 33)
        Me.lblInstruct.TabIndex = 4
        Me.lblInstruct.Text = "Input a TTL high or low level to auxillary digital inputs to change Data Value."
        Me.lblInstruct.TextAlign = System.Drawing.ContentAlignment.TopCenter
        '
        'lblDemoFunction
        '
        Me.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window
        Me.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDemoFunction.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDemoFunction.Location = New System.Drawing.Point(40, 16)
        Me.lblDemoFunction.Name = "lblDemoFunction"
        Me.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDemoFunction.Size = New System.Drawing.Size(241, 41)
        Me.lblDemoFunction.TabIndex = 0
        Me.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.DIn() and AuxPort"
        Me.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter
        '
        'frmDigAuxIn
        '
        Me.AcceptButton = Me.cmdEndProgram
        Me.AutoScaleBaseSize = New System.Drawing.Size(6, 13)
        Me.BackColor = System.Drawing.SystemColors.Window
        Me.ClientSize = New System.Drawing.Size(331, 246)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.cmdEndProgram, Me.lblShowPortVal, Me.lblPortVal, Me.lblInstruct, Me.lblDemoFunction})
        Me.Font = New System.Drawing.Font("Arial", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.ForeColor = System.Drawing.SystemColors.WindowText
        Me.Location = New System.Drawing.Point(7, 103)
        Me.Name = "frmDigAuxIn"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.Manual
        Me.Text = "Universal Library Digital In (AUXPORT)"
        Me.ResumeLayout(False)

    End Sub
#End Region

#Region "Universal Library Initialization - Expand this region to change error handling, etc."

    Private Sub InitUL()
        Dim ULStat As MccDaq.ErrorInfo
        Dim devNum As Integer
        Dim portType As Integer
        Dim inmask, outmask As Integer

        ' Initiate error handling
        '  activating error handling will trap errors like
        '  bad channel numbers and non-configured conditions.
        '  Parameters:
        '    MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
        '    MccDaq.ErrorHandling.StopAll   :if any error is encountered, the program will stop
        ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
            Stop
        End If


        ' Determine if board has AuxPort
        '  Parameters:
        '      devNum      : 0-based digital device index
        '      PortType    : return value for type of device
        devNum = 0
        PortType = 0
        ULStat = DaqBoard.DioConfig.GetDevType(devNum, PortType)
        If (PortType <> MccDaq.DigitalPortType.AuxPort) Then
            ' ERROR: This board does NOT have an AuxPort
            tmrReadInputs.Stop()
            Stop
            End
        End If


        ' Check if AuxPort needs configuring 
        '  Parameters:
        '      devNum  : 0-based digital device index
        '      mask    : bitmask indicating corresponding bit is in direction of ConfigItem(DIINMASk or DIOUTMASK)
        inmask = 0
        outmask = 0
        ULStat = DaqBoard.DioConfig.GetDInMask(devNum, inmask)
        If (ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors) Then Stop

        ULStat = DaqBoard.DioConfig.GetDOutMask(devNum, outmask)
        If (ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors) Then Stop

        ' Only non-configurable AuxPort's have overlapping input and output bitmasks 
        If (0 = (inmask And outmask)) Then ULStat = DaqBoard.DConfigPort(portType, MccDaq.DigitalPortDirection.DigitalIn)

    End Sub
#End Region

End Class